src/hooks/common/useDebounce.test.tsx

```diff
@@ -0,0 +1,97 @@
+import { useState } from 'react';
+import { cleanup, fireEvent, render, screen } from '@testing-library/react';
+import { renderHook } from '@testing-library/react-hooks';
+import { afterEach, beforeEach, describe, expect, test, vi } from 'vitest';
+
+import useDebounce from './useDebounce';
+
+describe('hooks/common/useDebounce', () => {
+  test('정의되어 있는가', () => {
+    expect(useDebounce).toBeDefined();
+  });
+
+  test('함수를 반환하는가', () => {
+    const testFn = () => {
+      return;
+    };
+    const { result } = renderHook(() => useDebounce(testFn, 500));
+    expect(typeof result.current).toBe('function');
+  });
+
+  describe('debounce', () => {
+    const App = () => {
+      const [val, setVal] = useState(0);
+
+      const onDebouncedClick = useDebounce(() => {
+        setVal((prev) => prev + 1);
+      }, 500);
+
+      return (
+        <div>
+          <span data-testid="count">{val}</span>
+          <button data-testid="button" type="button" onClick={onDebouncedClick}>
+            click
+          </button>
+        </div>
+      );
+    };
+
+    beforeEach(() => {
+      vi.useFakeTimers();
+    });
+
+    afterEach(cleanup);
+
+    test('반환된 함수를 실행하면 ms이후 실행된다', async () => {
+      render(<App />);
+      const Button = screen.getByTestId('button');
+      const Count = screen.getByTestId('count');
+
+      fireEvent.click(Button);
+      await vi.advanceTimersByTime(500);
+
+      expect(Count.innerHTML).toBe('1');
+    });
```

Comment:
500ms 전에 실행 안된 것도 테스트하면 좋을 거 같아요 !

```suggestion
    test('반환된 함수를 실행하면 ms이후 실행된다', async () => {
      render(<App />);
      const Button = screen.getByTestId('button');
      const Count = screen.getByTestId('count');

      fireEvent.click(Button);
      expect(Count.innerHTML).toBe('0');
      
      await vi.advanceTimersByTime(500);

      expect(Count.innerHTML).toBe('1');
    });
```